Typecast numeric cell values to string in HtmlHelper::_renderCells()

YveIce opened the pull request:
This patch typecast numeric cell values to string in HtmlHelper::_renderCells() before passing it to HtmlHelper::tableCell(), while strict typehint "String" for HtmlHelper::tableCell() stays untouched.
I personally would avoid to check if method "__toString()" exists if an class/object is passed, at the time of PHP-7.x, on a later PHP-8.x only release Uniontype typehint like "stringable|string|int|float" on HtmlHelper::tableCell() will do the trick for us in a much better way. 
On type bool, everyone has his personal opinion. (And i hate flamewar). My thoughts are, that common functions like substr() return (bool) false if they fail. If we typecast them silent, users may not take notice of misbehave of there code as no exception is thrown. 

Resolves https://github.com/cakephp/cakephp/issues/14921

YveIce committed:
to: https://github.com/cakephp/cakephp/issues/14921

YveIce commented:
And what is with:
```php
function tostring($value):string
{
   return (String) $value."<br>\n";
}
echo "bool(false) => ".tostring(false); // => returns empty string
echo "bool(true) => ".tostring(true); // => returns '1'
```
??? 
IMHO that's not a good behave.

dereuromark commented:
If you pass bool, that is what you get here for all stringish operations. So this is to be expected and OK IMO.
If you need to handle bool differently, do that before passing it on to any of those methods.

dereuromark commented on src/View/Helper/HtmlHelper.php:
```diff
@@ -790,6 +790,10 @@ protected function _renderCells(array $line, bool $useCount = false): array
             }
 
+            if(!is_string($cell) && is_numeric($cell)) {
+                $cell = (string) $cell;
```
$cell = (string)$cell;

YveIce committed:
[public function tableCells()](https://github.com/cakephp/cakephp/blob/8fb58b0889e8d6f1bcfdd31d6d27bc3f8cb2ac25/src/View/Helper/HtmlHelper.php#L724) has no typehint(array|string as PHP-Doc Tag) on argument $data. If a String is passed, it will first be warped as an array and later it will be unpacked in a foreach loop and passed as string to [HtmlHelper::_renderCells()](https://github.com/cakephp/cakephp/blob/8fb58b0889e8d6f1bcfdd31d6d27bc3f8cb2ac25/src/View/Helper/HtmlHelper.php#L754) but _renderCells() has a typehint and aspects the argument $line to be an array.

Double warp $data if it is of type String in tableCells(), so it can be passed as array to _renderCells()

ADmad commented:
Fixed in #14940
